app/src/main/java/kth/jjve/xfran/repositories/UserRepo.java

```diff
@@ -84,6 +84,40 @@ public static UserRepo getInstance() {
         return up;
     }
 
+    public MutableLiveData<String> getUserName(){
+        un = new MutableLiveData<>();
+        initFirebase();
+
+        if (firebaseAuth.getCurrentUser() != null){
+            //authenticate user
+            userID = firebaseAuth.getCurrentUser().getUid();
+
+            //get reference to user data
+            DocumentReference documentReference = firebaseFirestore.collection("users").document(userID);
+            documentReference.get().addOnCompleteListener(task -> {
+                if (task.isSuccessful()) {
+                    DocumentSnapshot document = task.getResult();
+                    assert document != null;
+                    // get username
+                    try {
+                        un.setValue(document.getString("firstName"));
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                        Log.i("UserRep", "No firstName found");
+                    }
+                }
+            });
+        } else {
+            un.setValue("not logged in");
+        }
+        return un;
+    }
+
+    private void initFirebase(){
+        firebaseAuth = FirebaseAuth.getInstance();
+        firebaseFirestore = FirebaseFirestore.getInstance();
+    }
```

Comment:
Could this be public? Bc it is used in other activities too

Reply:
Probably yes, but I wouldn't know how it would be handy. Would you mind if I don't do it in this PR and maybe check it out later?